src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -213,6 +225,16 @@ removeSubscription AgentClient {subscrConns, subscrSrvrs} connId = atomically $
       let cs' = S.delete connId cs
        in if S.null cs' then Nothing else Just cs'
 
+addActivation :: MonadUnliftIO m => AgentClient -> ConnId -> Async () -> m ()
+addActivation c connId a =
+  atomically $
+    modifyTVar (pendingActivations c) $ M.insert connId a
```

Comment:
no need to have them on 3 separate lines - it's short...

Reply:
also, it's more common to define such functions in STM monad and use atomically at call sites

Reply:
also - there should probably be check to prevent adding activation for the same connection - SUB can be used multiple times, currently it'll spawn activation every time SUB is used?

Reply:
TODO - check map and don't spawn thread if it exists

Reply:
done